OCastTests/TestOCastDomainGeneral.swift

```diff
@@ -0,0 +1,159 @@
+        let oCastApplicationLayer = OCastApplicationLayer(service: "service", data: oCastDataLayer)
+        let oCastDeviceLayer = OCastDeviceLayer(source: "source", destination: "destination", id: 1, status: "status", type: "type", message: oCastApplicationLayer)
+
+        jsonObject = oCastDeviceLayer
+    }
+    
+    override func tearDown() {
+    }
+    
+    func testDecodingWhenMissingId() throws {
+        assertThrowsKeyNotFound("id", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "id"))
+    }
+    
+    func testDecodingWhenMissingMessage() throws {
+        assertThrowsKeyNotFound("message", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "message"))
+    }
+    
+    func testDecodingWhenMissingType() throws {
+        assertThrowsKeyNotFound("type", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "type"))
+    }
+    
+    func testDecodingWhenMissingDestination() throws {
+        assertThrowsKeyNotFound("dst", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "dst"))
+    }
+    
+    func testDecodingWhenMissingSource() throws {
+        assertThrowsKeyNotFound("src", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "src"))
+    }
+    
+    func testDecodingWhenWringValue() throws {
+    }
+    
+    func testEncoding() throws {
+        let jsonDataTest = try! JSONEncoder().encode(jsonObject)
+        let jsonStringTest = String(data: jsonDataTest, encoding: .utf8)!
+        let trimmedTest = jsonStringTest.replacingOccurrences(of: "\\s*", with: "", options: .regularExpression)
+        
+        XCTAssertNotNil(trimmedTest.range(of: jsonRegexPattern, options: .regularExpression, range: nil, locale: nil))
+    }
+    
+    func testDecoding() {
+        let decoded = try! JSONDecoder().decode(OCastDeviceLayer<MediaTrackCommand>.self, from: jsonData)
+        XCTAssertTrue(decoded.isEqual(deviceLayer: jsonObject!))
+    }
+    
+    // Test that a missing key from the Json throwing an exception
+    private func assertThrowsKeyNotFound<T: Decodable>(_ expectedKey: String, decoding: T.Type, from data: Data, file: StaticString = #file, line: UInt = #line) {
+        XCTAssertThrowsError(try JSONDecoder().decode(decoding, from: data), file: file, line: line) { error in
+            if case .keyNotFound(let key, _)? = error as? DecodingError {
+                XCTAssertEqual(expectedKey, key.stringValue, "Expected missing key '\(key.stringValue)' to equal '\(expectedKey)'.", file: file, line: line)
+            } else {
+                XCTFail("Expected '.keyNotFound(\(expectedKey))' but got \(error)", file: file, line: line)
+            }
+        }
+    }
+}
+
+extension OCastDeviceLayer {
```

Comment:
If we extend a class that is tested,  does this not alter the behavior of the tested class?

Reply:
The purpose of these classes is to test the equality between two DeviceLayer objects. They do not alter the behavior of the tested class and are only used in the context of this test.